docs/dapp/diagrams/c10l-smart-contract-tx.mmd

```diff
@@ -0,0 +1,49 @@
+sequenceDiagram title: Confidential Smart Contract Transaction Flow
+    participant Client as Client<br/>(e.g. Node.js,<br/>MetaMask)
+    participant Web3 as Oasis Web3<br/>gateway
+    participant ClientN as Oasis Node<br/>ParaTime client
+    participant Compute as Oasis Node<br/>ParaTime compute<br/>🔒
+    participant KM as Oasis Node<br/>Key manager<br/>🔒
+
+    Client->>Client: Create transaction<br/>body
+
+    opt Encrypt Transaction
+        Client->>+Web3: Get ephemeral<br/>public key R
+        Web3->>+ClientN: Get ephemeral<br/>public key R
+        ClientN->>+KM: Get ephemeral<br/>public key R
+        KM->>KM: Derive x25519 keypair R,R'<br/>from Master key ||<br/>Epoch || ParaTime ID
+        KM->>-ClientN: Ephemeral public key R
+        Note right of ClientN: R is cached<br/>until the end<br/>of epoch
+        ClientN->>-Web3: Ephemeral<br/>public key R
+        Web3->>-Client: Ephemeral<br/>public key R
+
+        Client->>Client: Generate x5519<br/>keypair C,C'
+        Client->>Client: Encrypt transaction body<br/>with Deoxys-II envelope<br/>using C',R
```

Comment:
This needs to be split into two steps:
* Derive shared secret `s := X25519(C', R)`
* Encrypt call data with Deoxys-II using shared secret s.

Also it is not the entire transaction body that is encrypted but just the call data.